src/app/Main.js

```diff
@@ -53,32 +54,53 @@ class Main extends Component {
     }
   }
 
+  setPublic = (state) => {
+    this.setState({
+      isPublic: state
+    })
+  }
+
+  
   uploadClose = () => {
     this.setState({
       open: false,
       body: false,
       fileName: false,
     })
   }
-
+  
   uploadClick = () => {
     this.setState({
       open: true,
     })
   }
 
+  doSetPublicTag = (name, state) => {
+    const { folder } = this.state
+    fetch(`${name}?public=${state === true ? "False" : "True"}`, {
+      method: "PUT",
+    })
+    .then(response => response.text())
+    .then(text => {
+      this.setState({ response: text })
+      console.log(text)
+      this.list(folder)
+    })
+  }
+  
   doUpload = () => {
-    const {file, folder, body} = this.state
+    const { file, folder, body, isPublic } = this.state
     if(file) {
-      fetch(file, {
+      fetch(`${file}?public=${isPublic ? "True" : "False"}`, {
```

Comment:
`isPublic ? "True" : "False"` is this better than doing isPublic.toString() and uppercasing the first letter?

Reply:
Both works, I thought this was clearer